lib/formatters/preprocessWarnings.mjs

```diff
@@ -7,9 +7,9 @@ import { DEFAULT_SEVERITY, SEVERITY_ERROR } from '../constants.mjs';
 
 /**
  * Preprocess warnings in a given lint result.
- * Note that this function has a side-effect.
  *
  * @param {LintResult} result
+ * @modifies {result}
```

Comment:
see jsdoc/jsdoc#1709

Reply:
This seems unrelated.

Reply:
see
1. https://github.com/stylelint/stylelint/pull/8334#discussion_r1929527807
2. https://github.com/stylelint/stylelint/pull/8334/files#diff-6c2b58f14ee9189aabe9078bedf28bc0b5e62756820c2bd84422c8fc87a1096dR67